File: 2025/python/main.py
import sys
import importlib

def main():
    if len(sys.argv) < 2:
        print("Usage: python main.py <day>")
        sys.exit(1)
    
    day = sys.argv[1].zfill(2)
    
    input_path = None
    try:
        module = importlib.import_module(f"day{day}")
        input_path = f"../inputs/day{day}.txt"
        
        with open(input_path, "r") as f:
            input_data = f.read()
        
        print(f"Part 1: {module.part1(input_data)}")
        print(f"Part 2: {module.part2(input_data)}")
    
    except ModuleNotFoundError:
        print(f"Day {day} not implemented")
        sys.exit(1)
    except FileNotFoundError:
        print(f"Input file {input_path} not found")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: 2025/rust/src/main.rs
use std::env;
use std::fs;

mod day01;
mod day02;
// mod dayXX;

fn main() {
    let args: Vec<String> = env::args().collect();
    
    if args.len() < 2 {
        eprintln!("Usage: aoc <day>");
        std::process::exit(1);
    }

    let day = &args[1];
    let input_path = format!("../inputs/day0{}.txt", day);
    let input = fs::read_to_string(&input_path)
        .unwrap_or_else(|_| panic!("Could not read {}", input_path));

    match day.as_str() {
        "1" => println!("Part 1: {}\nPart 2: {}", day01::part1(&input), day01::part2(&input)),
        "2" => println!("Part 1: {}\nPart 2: {}", day02::part1(&input), day02::part2(&input)),
        // "XX" => println!("Part 1: {}\nPart 2: {}", dayXX::part1(&input), dayXX::part2(&input)),
        _ => eprintln!("Day {} not implemented", day),
    }
}
